client/fingerprint/network_unix.go

```diff
@@ -129,6 +133,54 @@ func (f *NetworkFingerprint) linkSpeedEthtool(path, device string) int {
 	return mbs
 }
 
+// ipAddress returns the first IPv4 address on the configured default interface
+// Tries Golang native functions and falls back onto ifconfig
+func (f *NetworkFingerprint) ipAddress(device string) string {
+	if ip, err := f.nativeIpAddress(device); err == nil {
```

Comment:
Well, no, that function should return the IP if there was no error, so if err == nil. Look at the if body.

If err is different than nil, its execution would continue, and we would try the ifconfig method.